### PR TITLE
Remove $ from comment adverts

### PR DESCRIPTION
**Pull request by arelra**
## What does this change?

- Remove $ from `comment-adverts`
- TypeScript `comment-adverts`, `dfp-env` and `get-advert-by-id`

Part of https://github.com/guardian/frontend/pull/23740

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## What is the value of this and can you measure success?

Remove old unsupported libraries (bonzo and qwery via $)
TypeScript all the things

### Tested

- [ ] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/comment-adverts.spec.ts [hunk 1]
```diff
@@ -133,10 +136,10 @@ describe('maybeUpgradeSlot', () => {
         const advert = {
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/comment-adverts.spec.ts:136]: Could you not declare `advert: Advert` here?

File: static/src/javascripts/projects/commercial/modules/comment-adverts.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+	ad.sizes.desktop.some(
+		(el) =>
+			(el[0] === 300 && el[1] === 600) ||
+			(el[0] === 160 && el[1] === 600),
+	);
+
+const maybeUpgradeSlot = (ad: Advert, $adSlot: Element): Advert => {
+	if (!containsDMPU(ad)) {
+		ad.sizes.desktop.push([300, 600], [160, 600]);
+        const defineSizeMappingFunction = ad.slot.defineSizeMapping as (asm: SizeMapping[]) => Slot;
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/comment-adverts.ts:69]: For consistency.

```suggestion
		const defineSizeMappingFunction = ad.slot.defineSizeMapping as (asm: SizeMapping[]) => Slot;
```

**Review by mxdvl — COMMENTED** [review]
🎉 Having the type for `DfpEnv` is going to be very useful!

Is there any hope that `$$` can help reduce type assertions in this case?

Please make sure the Flow equivalent of the TypeScript conversions are gone, too:

- static/src/javascripts.flow.archive/projects/commercial/modules/dfp/get-advert-by-id.js
- static/src/javascripts.flow.archive/projects/commercial/modules/dfp/dfp-env.js
- static/src/javascripts.flow.archive/projects/commercial/modules/comment-adverts.spec.ts
- static/src/javascripts.flow.archive/projects/commercial/modules/comment-adverts.js

File: static/src/javascripts/projects/commercial/modules/comment-adverts.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+		return Promise.resolve(false);
+	}
+
+	mediator.once('modules:comments:renderComments:rendered', () => {
+		const isLoggedIn = isUserLoggedIn();
+		const $commentMainColumn = document.querySelector<HTMLElement>(
+			'.js-comments .content__main-column',
+		);
+
+        if ($commentMainColumn) {
```
Comment (zekehuntergreen) [inline static/src/javascripts/projects/commercial/modules/comment-adverts.ts:112]: minor - is there a reason to add both an `if ($commentMainColumn)` block and the optional chaining [just below](if ($commentMainColumn))

File: static/src/javascripts/projects/commercial/modules/comment-adverts.spec.ts (hunk 1 above)
Comment (arelra) [reply static/src/javascripts/projects/commercial/modules/comment-adverts.spec.ts]: The problem here is we are creating a type which contains a subset of the `Advert` props for testing convenience.

But then passing it to the implementation which is expecting `Advert` even though it has missing properties.

I tried `Advert`, `Partial<Advert>` and extending `Advert` but ran into the same/further problems when using the object with the real implementation which is expecting `Advert`.

`Partial` does type checking on the properties but you still need to `(advert as Advert)` whenever using the object.

Given its a test convenience and it's not actually a real `Advert` this seemed like the easiest workaround.

I have however updated the test to use a function `createTestAdvert` which creates a `Partial` and then returns `as Advert` which shifts the problem to one function.

```js
const createTestAdvert = (testAdvert: Partial<Advert>): Advert =>
    ({ ... testAdvert } as Advert);
```

File: static/src/javascripts/projects/commercial/modules/comment-adverts.ts (hunk 3 above)
Comment (arelra) [reply static/src/javascripts/projects/commercial/modules/comment-adverts.ts:112]: Yep so the optional chaining will shortcut to `undefined` if it's not a valid chain.

In the next `then` function we try to use `$commentMainColumn` and TypeScript complains that this maybe undefined, hence the guard.

This looks like an oversight in the previous implementation as (I think) this code runs on every page so its possible for the the element `$commentMainColumn` to not exist.

**Comment by arelra** [comment]
> Please make sure the Flow equivalent of the TypeScript conversions are gone, too:
> 
> * static/src/javascripts.flow.archive/projects/commercial/modules/dfp/get-advert-by-id.js
> * static/src/javascripts.flow.archive/projects/commercial/modules/dfp/dfp-env.js
> * static/src/javascripts.flow.archive/projects/commercial/modules/comment-adverts.spec.ts
> * static/src/javascripts.flow.archive/projects/commercial/modules/comment-adverts.js

Was on the list... have removed 👍

**Comment by arelra** [comment]
> Is there any hope that `$$` can help reduce type assertions in this case?

Having removed $, the implementation now expects `Element` types.

I'm always divided by $$ and its utility, as a replacement for $ and a very limited set of bonzo functions (`css` etc) I can understand. But just to encapsulate `querySelectorAll` (and not `querySelector`) it seems so close to native that it is feels superfluous to me.

We can discuss, for now I've added a helper function to the test case to remove the bulk of the repetitive assertions.

File: static/src/javascripts/projects/commercial/modules/comment-adverts.spec.ts (hunk 1 above)
Comment (ealitten) [reply static/src/javascripts/projects/commercial/modules/comment-adverts.spec.ts]: Yeah I had the same issues with Advert.test.ts - this kind of issue must come up a lot, maybe someone has found a better way to deal with creating partial objects as mocks when testing but keeping TS happy

**Review by mxdvl — APPROVED** [review]

**Pull request merged**